### PR TITLE
use postcss-plugins to complete plugin data at prebuild

### PR DESCRIPTION
**Pull request by sylvainpolletvillard**
To avoid adding the entire plugins.json to the bundle, I retrieve and complete the data at build time, just like the way you build the README list.

Now we have a preferred name and URL for each plugin, in addition to a description, the author name, the stars count and some tags. I did not use the new data except the description, but it would be probably useful when redesigning the plugins list.

### CODE REVIEW

**Comment by RyanZim** [comment]
Looks good overall; want to build and try locally before merging.

File: plugins.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  {
+    name: 'rucksack',
+    pkgName: 'rucksack-css',
+  },
+  {
+    name: 'lost',
+  },
+  {
+    name: 'cssnext',
+    pkgName: 'postcss-cssnext',
```
Comment (RyanZim) [inline plugins.js]: The name here should be `postcss-cssnext`. `cssnext` refers to an older project by the same author that did the same thing, but was _not_ a postcss plugin. Ref: http://cssnext.io/postcss/

File: plugins.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
+module.exports = [
+  {
+    name: 'postcss-simple-vars',
+  },
+  {
+    name: 'postcss-advanced-variables',
+    config: { disable: ['@import'] },
+    url: 'https://github.com/jonathantneal/postcss-advanced-variables',
```
Comment (RyanZim) [inline plugins.js]: I take it this plugin is not on https://github.com/himynameisdave/postcss-plugins? If so, we should probably just add it to the list there.

This isn't a blocker for merging; just a note that we should do this sometime.

File: plugins.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  },
+  {
+    name: 'postcss-nesting',
+  },
+  {
+    name: 'postcss-nested',
+  },
+  {
+    name: 'short',
+    pkgName: 'postcss-short',
```
Comment (RyanZim) [inline plugins.js]: I'd personally prefer calling it `postcss-short`. The repo, website, and npm package use `postcss-short`, but the readme uses `short`.

@jonathantneal What's the correct name for displaying it on the REPL?

**Comment by RyanZim** [comment]
Also another problem I just noticed. The script outputs `app/plugins.js` in a way that doesn't comply with `prettier`'s rules. I'm not sure what the best solution is here, but we can't have the `build` and `format` scripts un-doing each other.

File: plugins.js (hunk 2 above)
Comment (sylvainpolletvillard) [reply plugins.js]: yes indeed. If @jonathanneal does not want it to be on postcss.parts for some reason, we can always put the data just here as well. I let him decide which name/description he wants for this

File: plugins.js (hunk 3 above)
Comment (sylvainpolletvillard) [reply plugins.js]: name is used to match with postcss-plugins data. We can always have a third property preferredName, but I think it should be the same than the one displayed on postcss.parts website, don't you think ?

File: plugins.js (hunk 1 above)
Comment (sylvainpolletvillard) [reply plugins.js]: same than previous comment. What should we do ? preferredName ?

File: plugins.js (hunk 3 above)
Comment (RyanZim) [reply plugins.js]: OK, if it's that way on postcss-parts, then leave it that way.

File: plugins.js (hunk 1 above)
Comment (RyanZim) [reply plugins.js]: Man, IDK, this is a mess.

File: build-plugins-list.js [hunk 4]
```diff
@@ -59,5 +61,5 @@ ${list}
 
 $3 END AUTO-GENERATED $4`
   );
-  fs.writeFileSync(file, newText);
+  fs.writeFileSync(file, prettier.format(newText, { parser }));
```
Comment (RyanZim) [inline build-plugins-list.js]: Need to pass in correct prettier options here; I have a config in the package.json to use single quotes, etc. Probably should use https://prettier.io/docs/en/api.html#prettierresolveconfigfilepath-options to get the options.

Comment (sylvainpolletvillard) [reply build-plugins-list.js]: ah, I was searching what was the problem. I assumed the config was resolved automatically, but we're not in the #0cjs era yet :)

Comment (RyanZim) [reply build-plugins-list.js]: Yeah, for some reason the API doesn't resolve the config.

File: plugins.js (hunk 2 above)
Comment (sylvainpolletvillard) [reply plugins.js]: oops, sorry for the bother. I mean @jonathantneal

Comment (jonathantneal) [reply plugins.js]: I would want it there. I just forgot to add plugins after a certain time.

File: plugins.js (hunk 3 above)
Comment (jonathantneal) [reply plugins.js]: Yes, I should update the README to be `postcss-short`. Amazing low priority unless someone wants to file an issue. 😄

File: plugins.js (hunk 1 above)
Comment (jonathantneal) [reply plugins.js]: The npm users have spoken. cssnext has 14,738 in the last month. postcss-cssnext has 617,906. There is no github project for cssnext, as it was renamed to postcss-cssnext.

Long live postcss-cssnext.

File: plugins.js (hunk 3 above)
Comment (RyanZim) [reply plugins.js]: PR done: https://github.com/jonathantneal/postcss-short/pull/14

File: plugins.js (hunk 1 above)
Comment (RyanZim) [reply plugins.js]: Refs https://github.com/himynameisdave/postcss-plugins/issues/229

File: plugins.js (hunk 2 above)
Comment (sylvainpolletvillard) [reply plugins.js]: PR done https://github.com/himynameisdave/postcss-plugins/pull/231

**Comment by RyanZim** [comment]
Please fix merge conflicts and I'll merge this.

**Comment by sylvainpolletvillard** [comment]
Okay, the last blocker for this PR is postcss-short name. 

Or "short" if you like to keep it... short

that's all for me

**Comment by RyanZim** [comment]
We can fix that later.

**Pull request merged**